docs/3.0-migration-guide.d/remove_MBEDTLS_X509_CHECK_x_KEY_USAGE_options.md

```diff
@@ -0,0 +1,20 @@
+Remove `MBEDTLS_X509_CHECK_*_KEY_USAGE` options from `config.h`
+--
+
+This change affects users who have chosen the compilation time options to disable
+the library's verification of the `keyUsage` and `extendedKeyUsage` fields of an x509
+certificate.
+
+The change is to remove MBEDTLS_X509_CHECK_KEY_USAGE and
+MBEDTLS_X509_CHECK_EXTENDED_KEY_USAGE from the configuration.
+
+After the change the options are removed and the compilation is done in a way that
+the verification of the key usage fields is allways enabled by default.
+
```

Comment:
Instead of the two following paragraphs I'd suggest:
"The verification of the `keyUsage` and `extendedKeyUsage` fields is important, disabling it can cause security issues and it is thus not recommended. If the verification is for some reason undesirable, it can still be disabled by means of the verification callback function passed to `mbedtls_x509_crt_verify()` (see the documentation of this function for more information)."